packages/design-system/src/components/TextField/Mask.jsx

```diff
@@ -69,33 +47,38 @@ function toDigitsAndAsterisks(value) {
 }
 
 /**
- * Remove all non-digits
+ * Performs various transforms to format provided string as currency.
  * @param {String} value
  * @returns {String}
  */
-function toDigits(value) {
-  return value.replace(/[^\d]/g, '');
-}
-
-/**
- * Convert string into a number (positive or negative float or integer)
- * @param {String} value
- * @returns {Number}
- */
-function toNumber(value) {
-  const sign = value.charAt(0) === '-' ? -1 : 1;
-  const parts = value.split('.');
-  // This assumes if the user adds a "." it should be a float. If we want it to
-  // evaluate as an integer if there are no digits beyond the decimal, then we
-  // can change it.
-  const hasDecimal = parts[1] !== undefined;
-  if (hasDecimal) {
-    const a = toDigits(parts[0]);
-    const b = toDigits(parts[1]);
-    return sign * parseFloat(`${a}.${b}`);
-  } else {
-    return sign * parseInt(toDigits(parts[0]));
+function toCurrency(value) {
```

Comment:
Even with all the transform steps, you've made this function really easy to follow.

Reply:
agreed! this is a hugeee improvement